test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js

```diff
@@ -3,865 +3,267 @@
-          name: 'new-service-dev-func1',
-          role: 'some:aws:arn:xx1:*:*',
-          vpc: {
-            securityGroupIds: ['xxx'],
-            subnetIds: ['xxx'],
-          },
-        },
-      };
-
-      return awsPackage
-        .mergeIamTemplates()
-        .then(
-          () =>
-            expect(
-              awsPackage.serverless.service.provider.compiledCloudFormationTemplate.Resources[
-                awsPackage.provider.naming.getRoleLogicalId()
-              ]
-            ).to.not.exist
-        );
-    });
-  });
-});
-
-describe('lib/plugins/aws/package/lib/mergeIamTemplates.test.js', () => {
-  describe.skip('TODO: No default role', () => {
-    it('should not create role resource if there are no functions', async () => {
-      await runServerless({ fixture: 'aws' });
-      // Replaces
-      // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L40-L49
-    });
 
-    it('should not create role resource with `provider.role`', async () => {
-      await runServerless({ fixture: 'function' });
-      // Replaces
-      // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L613-L636
-    });
-
-    it('should not create role resource with all functions having `functions[].role`', async () => {
-      await runServerless({ fixture: 'function' });
-      // Replaces
-      // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L638-L662
-      // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L747-L771
+      const IamRoleLambdaExecution = awsNaming.getRoleLogicalId();
+      const resourceIam = cfTemplate.Resources[IamRoleLambdaExecution];
+      expect(resourceIam).to.be.undefined;
     });
   });
 
   describe('Default role', () => {
-    describe.skip('TODO: Defaults', () => {
+    describe('Defaults', () => {
+      let naming;
+      let cfResources;
+
       before(async () => {
         await runServerless({
           fixture: 'function',
+          cliArgs: ['package'],
           configExt: {
-            // Configure one of the functions with custom "role", that will replace:
-            // https://github.com/serverless/serverless/blob/d8527d8b57e7e5f0b94ba704d9f53adb34298d99/lib/plugins/aws/package/lib/mergeIamTemplates.test.js#L587-L611
+            service: 'another-service',
+            functions: {
+              myFunction: {
+                handler: 'index.handler',
+              },
+              myFunctionWithRole: {
+                name: 'myCustomName',
+                handler: 'index.handler',
+                role: 'myCustRole0',
+              },
+            },
           },
+        }).then(({ cfTemplate, awsNaming }) => {
```

Comment:
Let's stick to just async/await syntax (no `.then` calls), here and in all other places